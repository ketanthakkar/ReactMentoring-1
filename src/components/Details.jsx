import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const DetailsWrapper = styled.div`
    display: flex;
    position: relative;
    align-items: flex-start;
    justify-content: flex-start;
    font-size: 18px;
`;

const DetailsSearchButton = styled.button`
    position: absolute;
    right: 0;
    top: -20px;
    border: none;
    border-radius: 3px;
    background-color: #fff;
    color: #f55263;
    text-transform: uppercase;
    font-weight: bold;
    padding: 10px 30px;
    cursor: pointer;
    display: block;
    z-index: 1;
    
    &:hover {
        color: #fff;
        background-color: #f55263;
    }
`;

const DetailsImage = styled.img`
    height: 400px;
    padding-top: 30px;
    margin: 0 50px 50px 0;
`;

const DetailsDescription = styled.div`
    color: #ffffff;
`;

const DetailsTitle = styled.div`
    font-size: 40px;
    font-weight: bold;
    color: #f55263;
    margin-top: 40px;
    
    &::after {
        content: '${props => props.vote_average}';
        display: inline-block;
        position: absolute;
        margin-left: 10px;
        font-size: 20px;
        color: #84ba67;
        border: 1px solid #84ba67;
        border-radius: 28px;
        width: 40px;
        height: 40px;
        line-height: 40px;
        text-align: center;
    }
`;

const DetailsTagline = styled.div`
    color: #bdbdbd;
    margin: 15px 0;
`;

const DetailsYear = styled.span`
    font-size: 24px;
    margin-right: 25px;
`;

const DetailsRuntime = styled.span`
    font-size: 24px;
`;

const DetailsOverview = styled.div`
    color: #bdbdbd;
    font-size: 22px;
    margin: 20px 0;
`;

class Details extends React.Component {
    constructor(props) {
        super(props)
    }

    _goToSearchPage() {
        this.props.goToSearchPage();
    }

    render() {
        const {
            item
        } = this.props;

        const {
            poster_path,
            title,
            vote_average,
            tagline,
            release_date,
            runtime,
            overview
        } = item;

        return (
            <DetailsWrapper>
                <DetailsSearchButton onClick={this._goToSearchPage.bind(this)}>
                    search
                </DetailsSearchButton>
                <DetailsImage
                    src={poster_path}
                />
                <DetailsDescription>
                    <DetailsTitle vote_average={vote_average}>{title}</DetailsTitle>
                    <DetailsTagline>{tagline}</DetailsTagline>
                    <div>
                        <DetailsYear>
                            {release_date && release_date.slice(0, release_date.indexOf('-'))}
                        </DetailsYear>
                        <DetailsRuntime>{runtime} min</DetailsRuntime>
                    </div>
                    <DetailsOverview>{overview}</DetailsOverview>
                </DetailsDescription>
            </DetailsWrapper>
        )
    }
}

Details.propTypes = {
    item: PropTypes.object,
    poster_path: PropTypes.string,
    title: PropTypes.string,
    vote_average: PropTypes.number,
    tagline: PropTypes.string,
    release_date: PropTypes.string,
    runtime: PropTypes.number,
    overview: PropTypes.string,
    goToSearchPage: PropTypes.func
};

Details.defaultProps = {
    item: {},
    poster_path: '',
    title: '',
    vote_average: 0,
    tagline: '',
    release_date: '',
    runtime: null,
    overview: '',
    goToSearchPage: null
};

export default Details;