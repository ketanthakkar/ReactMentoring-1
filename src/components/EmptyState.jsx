import React from 'react';
import styled from 'styled-components';

const EmptyResults = styled.div`
    font-size: 50px;
    height: 400px;
    padding-top: 100px;
    width: 100%;
    text-align: center;
`;

export class EmptyState extends React.Component {
    render() {
        return (
            <EmptyResults>No films found</EmptyResults>
        )
    }
}

export default EmptyState;
