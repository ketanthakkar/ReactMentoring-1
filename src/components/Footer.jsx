import React from 'react';
import styled from 'styled-components';

import Logo from './Logo';

const FooterSection = styled.div`
    background-color: #414141;
    height: 50px;
`;

const FooterWrapper = styled.div`
    display: flex;
    justify-content: space-between;
    align-items: center;
    width: 900px;
    height: 100%;
    margin: 0 auto;
`;

class Footer extends React.Component {
    constructor(props) {
        super(props)
    }

    render() {
        return (
            <FooterSection>
                <FooterWrapper>
                    <Logo/>
                </FooterWrapper>
            </FooterSection>
        )
    }
}

export default Footer;