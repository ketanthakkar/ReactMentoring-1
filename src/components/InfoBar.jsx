import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import APP_STATES from '../constants/APP_STATES';
import SORT_BY from '../constants/SORT_BY';

const InfoBarSection = styled.div`
    display: block;
    width: 100%;
    height: 50px;
    background-color: #f5f5f5;
`;

const InfoBarWrapper = styled.div`
    display: flex;
    align-items: center;
    justify-content: space-between;
    height: 50px;
    width: 900px;
    margin: 0 auto;
`;

const InfoBarLabel = styled.label`
    color: #0e0e0e;
`;

const InfoBarSortByLink = styled.span`
    margin: 5px 5px 5px 20px;
    color: #0e0e0e;
    text-transform: none;
    
    &:hover {
        color:#f55263;
        cursor: pointer;
    }
    
    &.active {
        color:#f55263;
    }
`;

class InfoBar extends React.Component {

    onSortByReleaseDateClick() {
        this.props.setSortBy(SORT_BY.RELEASE_DATE);
        this.props.getItems();
    }

    onSortByRatingClick() {
        this.props.setSortBy(SORT_BY.RATING);
        this.props.getItems();
    }

    render() {
        const {
            appState,
            items,
            item,
            total,
            sortBy,
        } = this.props;
        let output = {};

        switch (appState) {
            case APP_STATES.SEARCH_PAGE:
                output = (
                    <InfoBarSection>
                        {items && items.length > 0 &&
                        <InfoBarWrapper>
                            <InfoBarLabel>
                                {total} movies found
                            </InfoBarLabel>
                            <div>
                                <InfoBarLabel>
                                    Sort by
                                </InfoBarLabel>
                                <InfoBarSortByLink
                                    className={sortBy === SORT_BY.RELEASE_DATE && 'active'}
                                    onClick={this.onSortByReleaseDateClick.bind(this)}
                                >
                                    release date
                                </InfoBarSortByLink>
                                <InfoBarSortByLink
                                    className={sortBy === SORT_BY.RATING && 'active'}
                                    onClick={this.onSortByRatingClick.bind(this)}
                                >
                                    rating
                                </InfoBarSortByLink>
                            </div>
                        </InfoBarWrapper>
                        }
                    </InfoBarSection>
                );
                break;
            case APP_STATES.DETAILS_PAGE:
                output = (
                    <InfoBarSection>
                        {items && items.length > 0 &&
                        <InfoBarWrapper>
                            <InfoBarLabel>
                                Films by {item && item.genres && item.genres[0]}
                            </InfoBarLabel>
                        </InfoBarWrapper>
                        }
                    </InfoBarSection>
                );
                break;
            default:
        }
        return (
            output
        )
    }
}

InfoBar.propTypes = {
    appState: PropTypes.string,
    items: PropTypes.array,
    item: PropTypes.object,
    total: PropTypes.number,
    sortBy: PropTypes.string,
    setSortBy: PropTypes.func,
};

InfoBar.defaultProps = {
    appState: APP_STATES.SEARCH_PAGE,
    items: [],
    item: {},
    total: 0,
    setSortBy: null,
};

export default InfoBar;
