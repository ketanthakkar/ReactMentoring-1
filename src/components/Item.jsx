import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import {Link} from 'react-router-dom';

const ItemWrapper = styled.div`
    width: 250px;
    margin: 20px 75px 20px 0;
    
    :nth-child(3n) {
        margin: 20px 0 20px 0;
    }
`;

const ItemImage = styled.img`
    width: 250px;
    cursor: pointer;
`;

const ItemInfo = styled.div`
    display: flex;
    align-items: flex-start;
    justify-content: space-between;
    margin-top: 5px;
`;

const ItemTitle = styled.span`
    flex-basis: 70%;
`;

const ItemDate = styled.span`
    border: 1px solid gray;
    border-radius: 3px;
    padding: 1px 3px;
`;

const ItemGenres = styled.div`
    color: gray;
    margin-top: 5px;
    font-size: 14px;
`;

class Item extends React.Component {
    render() {
        const {
            item,
        } = this.props;

        const {
            poster_path,
            title,
            release_date,
            genres,
        } = item;

        return (
            <ItemWrapper>
                <Link to={`/film/${item.id}`}>
                    <ItemImage
                        src={poster_path}
                    />
                </Link>
                <ItemInfo>
                    <ItemTitle>{title}</ItemTitle>
                    <ItemDate>{release_date && release_date.slice(0, release_date.indexOf('-'))}</ItemDate>
                </ItemInfo>
                <ItemGenres>{genres && genres.join(' & ')}</ItemGenres>
            </ItemWrapper>
        )
    }
}

Item.propTypes = {
    item: PropTypes.object,
};

Item.defaultProps = {
    item: {},
};

export default Item;
