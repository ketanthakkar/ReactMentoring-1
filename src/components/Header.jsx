import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import Logo from './Logo';
import SearchField from './SearchField';
import InfoBar from './InfoBar';
import Details from './Details';

import background from '../img/background.jpg';

import APP_STATES from '../constants/APP_STATES';

const HeaderSection = styled.div`
    width: 100%;
    height: 100%;
    background-image: linear-gradient(rgba(0, 0, 0, 0.8), rgba(0, 0, 0, 0.4)), url(${background});
`;

const HeaderWrapper = styled.div`
    width: 900px;
    margin: 0 auto;
    padding-top: 20px;
`;

class Header extends React.Component {
    constructor(props) {
        super(props);
    }

    render() {
        const {
            appState,
            selectedItem,
            items,
            goToSearchPage
        } = this.props;

        return (
            <HeaderSection>
                <HeaderWrapper>
                    <Logo/>
                    {appState === APP_STATES.DETAILS_PAGE ?
                        <Details
                            item={items.find(item => item.id === selectedItem)}
                            goToSearchPage={goToSearchPage}
                        />
                        :
                        <SearchField {...this.props}/>
                    }
                </HeaderWrapper>
                <InfoBar {...this.props}/>
            </HeaderSection>
        )
    }
}

Header.propTypes = {
    appState: PropTypes.string,
    selectedItem: PropTypes.number,
    items: PropTypes.array,
    goToSearchPage: PropTypes.func
};

Header.defaultProps = {
    appState: '',
    selectedItem: null,
    items: [],
    goToSearchPage: null
};

export default Header;