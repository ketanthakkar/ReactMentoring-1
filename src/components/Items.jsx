import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import Item from './Item';

const ItemsSection = styled.div`
    background-color: #fff;
`;

const ItemsWrapper = styled.div`
    display: flex;
    justify-content: flex-start;
    flex-wrap: wrap;
    width: 900px;
    margin: 0 auto;
`;

const EmptyResults = styled.div`
    font-size: 50px;
    height: 400px;
    padding-top: 100px;
    width: 100%;
    text-align: center;
`;

class Items extends React.Component {
    render() {
        const {
            items,
            getItem,
            getItemsByGenre
        } = this.props;

        return (
            <ItemsSection>
                <ItemsWrapper>
                    {items && items.length ? (
                        items.map(
                            item => (
                                <Item
                                    key={item.id}
                                    item={item}
                                    getItem={getItem}
                                    getItemsByGenre={getItemsByGenre}
                                />
                            )
                        )
                    ) : (
                        <EmptyResults>No films found</EmptyResults>
                    )}
                </ItemsWrapper>
            </ItemsSection>
        )
    }
}

Items.propTypes = {
    items: PropTypes.array,
    getItem: PropTypes.func,
    getItemsByGenre: PropTypes.func,
};

Items.defaultProps = {
    items: [],
    getItem: null,
    getItemsByGenre: null,
};

export default Items;
