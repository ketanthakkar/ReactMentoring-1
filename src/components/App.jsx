import React from 'react';
import styled from 'styled-components';

import Header from './Header';
import Items from './Items';
import Footer from './Footer';
import ErrorBoundary from './ErrorBoundary';

import items from '../mock-data/items.json';

import APP_STATES from '../constants/APP_STATES';
import SEARCH_BY from '../constants/SEARCH_BY';
import SORT_BY from '../constants/SORT_BY';

const AppWrapper = styled.div`
    height: 100%;
    font-family: sans-serif;
`;

class App extends React.Component {
    constructor() {
        super();
        this.state = {
            value: '',
            searchBy: SEARCH_BY.TITLE,
            sortBy: SORT_BY.RELEASE_DATE,
            appState: APP_STATES.SEARCH_PAGE,
            selectedItem: null,
            items: items.data,
            // items: []
        }
    }

    goToSearchPage() {
        this.setState({
            appState: APP_STATES.SEARCH_PAGE
        })
    }

    onSearch(options) {
        const {
            value = this.state.value,
            searchBy = this.state.searchBy,
            sortBy = this.state.sortBy
        } = {...options};
        this.setState({
            value,
            searchBy,
            sortBy,
            selectedItem: null,
            appState: APP_STATES.SEARCH_PAGE
        });
        console.log(`value:${value} searchBy:${searchBy} sortBy:${sortBy}`);
    }

    onItemClick(id) {
        this.setState({
            selectedItem: id,
            appState: APP_STATES.DETAILS_PAGE
        });
        console.log(id);
    }

    render() {
        return (
            <AppWrapper>
                <ErrorBoundary>
                    <Header
                        {...this.state}
                        onSearch={this.onSearch.bind(this)}
                        goToSearchPage={this.goToSearchPage.bind(this)}
                    />
                    <Items
                        {...this.state}
                        onItemClick={this.onItemClick.bind(this)}
                    />
                    <Footer/>
                </ErrorBoundary>
            </AppWrapper>
        )
    }
}

export default App;
