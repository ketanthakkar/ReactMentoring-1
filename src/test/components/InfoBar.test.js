import React from 'react';
import { shallow } from 'enzyme';

import {InfoBar} from 'components/InfoBar';

describe('InfoBar', () => {
    const infoBar = shallow(<InfoBar />);

    it('should be like snapshot', () => {
        expect(infoBar).toMatchSnapshot();
    });
});
