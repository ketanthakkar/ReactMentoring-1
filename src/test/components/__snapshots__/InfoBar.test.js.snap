// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoBar should be like snapshot when AppState is DetailsPage 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <InfoBar
    appState="details page"
    item={Object {}}
    items={
      Array [
        Object {},
      ]
    }
    setSortBy={null}
    total={0}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <InfoBar__InfoBarWrapper>
        <InfoBar__InfoBarLabel>
          Films by 
        </InfoBar__InfoBarLabel>
      </InfoBar__InfoBarWrapper>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <InfoBar__InfoBarLabel>
          Films by 
        </InfoBar__InfoBarLabel>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            "Films by ",
            undefined,
          ],
        },
        "ref": null,
        "rendered": Array [
          "Films by ",
          undefined,
        ],
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <InfoBar__InfoBarWrapper>
          <InfoBar__InfoBarLabel>
            Films by 
          </InfoBar__InfoBarLabel>
        </InfoBar__InfoBarWrapper>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <InfoBar__InfoBarLabel>
            Films by 
          </InfoBar__InfoBarLabel>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              "Films by ",
              undefined,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Films by ",
            undefined,
          ],
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`InfoBar should be like snapshot when AppState is SearchPage 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <InfoBar
    appState="search page"
    item={Object {}}
    items={
      Array [
        Object {},
      ]
    }
    setSortBy={null}
    total={0}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <InfoBar__InfoBarWrapper>
        <InfoBar__InfoBarLabel>
          0
           movies found
        </InfoBar__InfoBarLabel>
        <div>
          <InfoBar__InfoBarLabel>
            Sort by
          </InfoBar__InfoBarLabel>
          <InfoBar__InfoBarSortByLink
            className={false}
            onClick={[Function]}
          >
            release date
          </InfoBar__InfoBarSortByLink>
          <InfoBar__InfoBarSortByLink
            className={false}
            onClick={[Function]}
          >
            rating
          </InfoBar__InfoBarSortByLink>
        </div>
      </InfoBar__InfoBarWrapper>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <InfoBar__InfoBarLabel>
            0
             movies found
          </InfoBar__InfoBarLabel>,
          <div>
            <InfoBar__InfoBarLabel>
              Sort by
            </InfoBar__InfoBarLabel>
            <InfoBar__InfoBarSortByLink
              className={false}
              onClick={[Function]}
            >
              release date
            </InfoBar__InfoBarSortByLink>
            <InfoBar__InfoBarSortByLink
              className={false}
              onClick={[Function]}
            >
              rating
            </InfoBar__InfoBarSortByLink>
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              0,
              " movies found",
            ],
          },
          "ref": null,
          "rendered": Array [
            0,
            " movies found",
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <InfoBar__InfoBarLabel>
                Sort by
              </InfoBar__InfoBarLabel>,
              <InfoBar__InfoBarSortByLink
                className={false}
                onClick={[Function]}
              >
                release date
              </InfoBar__InfoBarSortByLink>,
              <InfoBar__InfoBarSortByLink
                className={false}
                onClick={[Function]}
              >
                rating
              </InfoBar__InfoBarSortByLink>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Sort by",
              },
              "ref": null,
              "rendered": "Sort by",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "release date",
                "className": false,
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "release date",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "rating",
                "className": false,
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "rating",
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <InfoBar__InfoBarWrapper>
          <InfoBar__InfoBarLabel>
            0
             movies found
          </InfoBar__InfoBarLabel>
          <div>
            <InfoBar__InfoBarLabel>
              Sort by
            </InfoBar__InfoBarLabel>
            <InfoBar__InfoBarSortByLink
              className={false}
              onClick={[Function]}
            >
              release date
            </InfoBar__InfoBarSortByLink>
            <InfoBar__InfoBarSortByLink
              className={false}
              onClick={[Function]}
            >
              rating
            </InfoBar__InfoBarSortByLink>
          </div>
        </InfoBar__InfoBarWrapper>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <InfoBar__InfoBarLabel>
              0
               movies found
            </InfoBar__InfoBarLabel>,
            <div>
              <InfoBar__InfoBarLabel>
                Sort by
              </InfoBar__InfoBarLabel>
              <InfoBar__InfoBarSortByLink
                className={false}
                onClick={[Function]}
              >
                release date
              </InfoBar__InfoBarSortByLink>
              <InfoBar__InfoBarSortByLink
                className={false}
                onClick={[Function]}
              >
                rating
              </InfoBar__InfoBarSortByLink>
            </div>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                0,
                " movies found",
              ],
            },
            "ref": null,
            "rendered": Array [
              0,
              " movies found",
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <InfoBar__InfoBarLabel>
                  Sort by
                </InfoBar__InfoBarLabel>,
                <InfoBar__InfoBarSortByLink
                  className={false}
                  onClick={[Function]}
                >
                  release date
                </InfoBar__InfoBarSortByLink>,
                <InfoBar__InfoBarSortByLink
                  className={false}
                  onClick={[Function]}
                >
                  rating
                </InfoBar__InfoBarSortByLink>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Sort by",
                },
                "ref": null,
                "rendered": "Sort by",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "release date",
                  "className": false,
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "release date",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "rating",
                  "className": false,
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "rating",
                "type": [Function],
              },
            ],
            "type": "div",
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
